src/v2_0_1/helpers/validator.rs

```diff
@@ -1,19 +1,56 @@
+use std::sync::OnceLock;
+
 use regex::Regex;
 use validator::ValidationError;
 
+static REGEX: OnceLock<Regex> = OnceLock::new();
+
 /// Helper function to validate identifierString
 ///
 /// # identfierString
 /// This is a case-insensitive dataType and can only contain characters from the following
 /// character set: `a-z`, `A-Z`, `0-9`, `'*'`, `'-'`, `'_'`, `'='`, `':'`, `'+'`, `'|'`, `'@'`, `'.'`
 pub fn validate_identifier_string(s: &str) -> Result<(), ValidationError> {
     // regex for identifierString as defined by the specification
-    let re = Regex::new(r"[a-z]|[A-Z]|[0-9]|[+]|[*]|[-]|[=]|[:]|[0]|[|]|[@]|[.]").unwrap();
-
-    let res = re.is_match(s);
+    let res = REGEX
+        .get_or_init(|| Regex::new(r"^[a-zA-Z0-9*+=:|@._-]+$").unwrap())
```

Comment:
The + symbol match the preceding character one or more times. In the specification the identifierString might be 0-n in length. I think we should use a * here instead.

For example in CostUpdatedRequest the specification of v2.0.1 states in the field type column: identifierString[0..36]